### PR TITLE
ENH add pydantic based validation for the new YAML patch instructions

### PR DESCRIPTION
**Pull request by jaimergp**
Follow up to #498 

cc @beckermr

### CODE REVIEW

File: recipe/patch_yaml_utils.py [hunk 1]
```diff
@@ -260,7 +265,7 @@ def _apply_patch_yaml(patch_yaml, record, subdir, fn):
                 elif not depends and subk in record:
                     del record[subk]
 
-            elif k == "remove_track_feature":
+            elif k == "remove_track_features":
```
Comment (jaimergp) [inline recipe/patch_yaml_utils.py:268]: I renamed this field for consistency with the other instructions, where the convention seemed to be `{action}_{repodata-field}`.

Comment (beckermr) [reply recipe/patch_yaml_utils.py:268]: Yes thank you! This was bothering me yesterday!

**Review by beckermr — APPROVED** [review]

**Comment by beckermr** [comment]
This code seems super slow now @jaimergp. Do we have to validate as we patch, or can we only validate in the tests?

**Comment by beckermr** [comment]
Yes, we validate all yamls in the tests that run as part of the package build. This will catch validation issues as PRs are sent.

**Comment by beckermr** [comment]
Shall we merge @jaimergp?

**Comment by jaimergp** [comment]
I think that's ok. I'm a bit puzzled on the performance though. The whole test suite [took 0.69s to run](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=771794&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=986b1512-c876-5f92-0d81-ba851554a0a3&l=180) (and that includes validation, right?), but the patching [still takes ~7 minutes](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=771794&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=986b1512-c876-5f92-0d81-ba851554a0a3&l=197). Not sure that was a bottleneck, but if it works it works!

**Comment by beckermr** [comment]
We were validating each part of each yaml for every yaml and every record in the repodata json. So we were literally running the test suite hundreds of thousands of times.

**Comment by jaimergp** [comment]
> So we were literally running the test suite hundreds of thousands of times.

Ah ok, I see! Thanks for clarifying.

**Pull request merged**